### DISCUSSION TITLE
fix the 2 GraphQL endpoint problem

### DISCUSSION THREAD

**Discussion by jeremybradbury** [directus/directus#12444] [Feature Requests]
First of all this is a major GraphQL antipattern unless you're using Apollo Server's federation tools. 

Second, `users_me` is under /graphql/system and the rest of the data is not.

Switching between 2 graphql endpoints or allowing an SPA to use both is a massive problem, so this pattern kills most SPA projects that include auth and GraphQL.

The problem created is quite obvious, not sure what problem it has solved. 

A quick fix for my use case would be to add `users_me` to the primary graph when authentication is provided, just like how other permissions work.

However, we're still stuck building out 2 clients/graphs/caches for the 2 endpoints if we want to use anything from system in an end user app or anything from items in an admin app without using the SDK along with GraphQL.

Security and role based permissions should be able to apply to the entire graph. hide most system stuff from non admins all of it, all of it (besides meta like site name and colors) from anonymous users, ezpz. Perhaps make it into it's own top level instead of Query use System instead like how Apollo added Subscribe. or even just a system object resolver under Query, then just change /system queries to {system:} queries.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
